uber/models.py

```diff
@@ -2038,7 +2052,9 @@ class PromoCode(MagModel):
             func.replace(func.replace(func.lower(code), '-', ''), ' ', ''),
             unique=True),
         CheckConstraint(func.trim(code) != '',
-            name='ck_promo_code_non_empty_code'))
+            name='ck_promo_code_non_empty_code'),
+        CheckConstraint(or_(discount != None, uses_allowed != None),
+            name='ck_promo_code_no_unlimited_use_free_badge'))
```

Comment:
I wanted to put this in the database to prevent even the possibility of unlimited free promo codes existing. In retrospect, there's just so many ways to work around it, I don't think it's worth it. I think I'm actually going to remove this constraint and do it entirely in `model_checks.py`.